src/components/SQLEditor/SQLEditor.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import AceEditor from 'react-ace';
+import 'ace-builds/src-noconflict/mode-sql';
+import 'ace-builds/src-noconflict/theme-github';
+import 'ace-builds/src-min-noconflict/ext-language_tools';
+import { Button } from '@crate.io/crate-ui-components';
+
+type Params = {
+  execCallback: (editorContents: string) => void,
+}
+
+function SQLEditor({ execCallback }: Params) {
+  const [editorContents, setEditorContents] = useState('');
+
+
+  return <div>
+    <AceEditor
+      height="150px"
+      width="120%"
+      mode="sql"
+      theme="github"
+      fontSize={16}
+      highlightActiveLine
+      enableLiveAutocompletion
+      editorProps={{ $blockScrolling: true }}
+      commands={[
+        {
+          name: 'ctrlEnter',
+          bindKey: {
+            win: 'Ctrl-Enter',
+            mac: 'Command-Enter',
+            // @ts-expect-error type problem
+            linux: 'Ctrl-Enter',
+          },
+          exec: (editor) => execCallback(editor.getValue()),
+        },
+      ]}
+      setOptions={{
+        showLineNumbers: true,
+        tabSize: 4,
```

Comment:
```suggestion
        tabSize: 2,
```

Should typically be 2 for SQL imho